tests/common/test_typing.py

```diff
@@ -65,6 +68,50 @@ class MyDataclass:
 TOptionalUnionLiTyDi = Optional[Union[TTestTyDi, TTestLi]]
 
 
+def test_is_callable_type() -> None:
+    class AsyncClass:
+        async def __call__(self):
+            pass
+
+    class NotAsyncClass:
+        def __call__(self):
+            pass
+
+    class ClassNoCall:
+        pass
+
+    class ClassBaseCall(NotAsyncClass):
+        pass
+
+    async def async_func():
+        pass
+
+    def non_async_func():
+        pass
+
+    CallableType = Callable[[Any], Any]
+
+    # new type is a callable and will be ignored
+    ANewType = NewType("ANewType", str)
```

Comment:
Maybe also add some generic type aliases like `AnyDict = Dict[str, Any]` or `SomeString = Optional[str]`

Reply:
we have them in common typing. and honestly we should probably ditch them...